server/src/utils/lifecycle.ts

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env node
+import { OpenAPIObject } from '@nestjs/swagger';
+import { SchemaObject } from '@nestjs/swagger/dist/interfaces/open-api-spec.interface';
+import { readFileSync } from 'node:fs';
+import { resolve } from 'node:path';
+import { Version } from 'src/utils/version';
+
+const outputPath = resolve(process.cwd(), '../open-api/immich-openapi-specs.json');
+const spec = JSON.parse(readFileSync(outputPath).toString()) as OpenAPIObject;
+
+type Items = {
+  oldEndpoints: Endpoint[];
+  newEndpoints: Endpoint[];
+  oldProperties: Property[];
+  newProperties: Property[];
+};
+type Endpoint = { url: string; method: string; endpoint: any };
+type Property = { schema: string; property: string };
+
+const metadata: Record<string, Items> = {};
+const trackVersion = (version: string) => {
+  if (!metadata[version]) {
+    metadata[version] = {
+      oldEndpoints: [],
+      newEndpoints: [],
+      oldProperties: [],
+      newProperties: [],
+    };
+  }
+  return metadata[version];
+};
+
+for (const [url, methods] of Object.entries(spec.paths)) {
+  for (const [method, endpoint] of Object.entries(methods) as Array<[string, any]>) {
+    const deprecatedAt = endpoint['x-immich-lifecycle']?.deprecatedAt;
+    if (deprecatedAt) {
+      trackVersion(deprecatedAt).oldEndpoints.push({ url, method, endpoint });
+    }
+
+    const addedAt = endpoint['x-immich-lifecycle']?.addedAt;
+    if (addedAt) {
+      trackVersion(addedAt).newEndpoints.push({ url, method, endpoint });
+    }
+  }
+}
+
+for (const [schemaName, schema] of Object.entries(spec.components?.schemas || {})) {
+  for (const [propertyName, property] of Object.entries((schema as SchemaObject).properties || {})) {
+    const propertySchema = property as SchemaObject;
+    if (propertySchema.description?.startsWith('This property was deprecated in')) {
```

Comment:
`This property was deprecated in` and `This property was added in` can also be constants and used in the decorator.